test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -0,0 +1,255 @@
+    for (ucp_lane_index_t lane = 0; lane < ucp_ep_num_lanes(ep()); ++lane) {
+        UCS_STATIC_BITMAP_SET(&tl_bitmap, ucp_ep_get_rsc_index(ep(), lane));
+    }
+
+    return tl_bitmap;
+}
+
+void test_ucp_ep_reconfigure::entity::connect(const ucp_test_base::entity *other,
+                                              const ucp_ep_params_t &ep_params,
+                                              int ep_idx, int do_set_ep)
+{
+    auto r_other     = static_cast<const entity*>(other);
+    auto worker_addr = r_other->get_address(ucp_tl_bitmap_max);
+    ucp_tl_bitmap_t tl_bitmap;
+    unsigned addr_indices[UCP_MAX_LANES];
+    ucp_ep_h ucp_ep;
+
+    tl_bitmap = m_exclude_ifaces ?
+                        UCS_STATIC_BITMAP_NOT(r_other->ep_tl_bitmap()) :
+                        ucp_tl_bitmap_max;
+
+    UCS_ASYNC_BLOCK(&worker()->async);
+    ASSERT_UCS_OK(ucp_ep_create_to_worker_addr(worker(), &tl_bitmap,
+                                               &worker_addr->second,
+                                               UCP_EP_INIT_CREATE_AM_LANE,
+                                               "reconfigure test", addr_indices,
+                                               &ucp_ep));
+
+    ucp_ep->conn_sn = 0;
+    ASSERT_TRUE(ucp_ep_match_insert(worker(), ucp_ep, worker_addr->second.uuid,
+                                    ucp_ep->conn_sn, UCS_CONN_MATCH_QUEUE_EXP));
+    m_workers[0].second.push_back(
+            ucs::handle<ucp_ep_h, ucp_test_base::entity*>(ucp_ep,
+                                                          ucp_ep_destroy));
+
+    ASSERT_UCS_OK(ucp_wireup_send_request(ucp_ep));
+
+    store_config();
+    UCS_ASYNC_UNBLOCK(&worker()->async);
+}
+
+bool test_ucp_ep_reconfigure::entity::has_matching_lane(
+        ucp_ep_h ep, ucp_lane_index_t lane_idx, const entity &other) const
+{
+    const auto lane     = &ucp_ep_config(ep)->key.lanes[lane_idx];
+    const auto resource = &ucph()->tl_rscs[ucp_ep_get_rsc_index(ep, lane_idx)];
+    auto addr           = other.get_address(other.ep_tl_bitmap());
+    const ucp_address_entry_t *ae;
+
+    ucs_carray_for_each(ae, addr->second.address_list,
+                        addr->second.address_count) {
+        if ((resource->tl_name_csum == ae->tl_name_csum) &&
+            ucp_wireup_is_lane_connected(ep, lane_idx, ae)) {
+            EXPECT_EQ(ae->sys_dev, lane->dst_sys_dev);
+            EXPECT_EQ(ae->md_index, lane->dst_md_index);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void test_ucp_ep_reconfigure::entity::verify_configuration(
+        const entity &other) const
+{
+    unsigned reused                  = 0;
+    const ucp_lane_index_t num_lanes = ucp_ep_num_lanes(ep());
+
+    EXPECT_EQ(num_lanes, ucp_ep_num_lanes(other.ep()));
+
+    for (ucp_lane_index_t lane = 0; lane < num_lanes; ++lane) {
+        /* Verify local and remote lanes are identical */
+        EXPECT_TRUE(has_matching_lane(ep(), lane, other));

```

Comment:
do we really need this? isn't it enough to test that communication send/recv is working?
same for checking num_lanes

Reply:
how is it possible to check all lanes without using multiple protocols/message sizes?

Reply:
IMO we can do basic test like active message or tag match on several message sizes and extend it
since communication ability is something we need to test anyway to ensure configuration is working

Reply:
maybe better as a complement?

Reply:
I see, so removed the num_lanes check